axelor-quality/src/main/java/com/axelor/apps/quality/service/QIIdentificationServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2005-2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.quality.service;
+
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.apps.quality.db.QIIdentification;
+import com.axelor.auth.AuthUtils;
+import com.axelor.common.StringUtils;
+import com.google.inject.Inject;
+import java.math.BigDecimal;
+
+public class QIIdentificationServiceImpl implements QIIdentificationService {
+  @Inject AppBaseService appBaseService;
```

Comment:
Missing protected

Also use injection with constructor as it is the preferred way in AOS